pkg/metallb/frr.go

```diff
@@ -0,0 +1,583 @@
+package metallb
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"time"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	frrtypes "github.com/openshift-kni/eco-goinfra/pkg/schemes/metallb/frrtypes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtimeClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// FrrConfigurationBuilder provides struct for the FrrConfiguration object containing connection to
+// the cluster and the FrrConfiguration definitions.
+type FrrConfigurationBuilder struct {
+	Definition *frrtypes.FRRConfiguration
+	Object     *frrtypes.FRRConfiguration
+	apiClient  runtimeClient.Client
+	errorMsg   string
+}
+
+// NewFrrConfigurationBuilder creates a new instance of FRRConfiguration.
+func NewFrrConfigurationBuilder(
+	apiClient *clients.Settings, name, nsname string) *FrrConfigurationBuilder {
+	glog.V(100).Infof(
+		"Initializing new Frrconfiguration structure with the following params: %s, %s",
+		name, nsname)
+
```

Comment:
oh also add a check and return nil if apiClient is nil as well as update the unit test for it please

Reply:
added